core/src/main/java/org/javarosa/core/model/instance/DataInstance.java

```diff
@@ -320,4 +320,8 @@ public CacheHost getCacheHost() {
     public void setCacheHost(CacheHost cacheHost) {
         this.mCacheHost = cacheHost;
     }
+
+    public void cleanCache() {
+        referenceCache = new CacheTable<TreeReference, T>();

```

Comment:
Can you nuke this cache by doing, like, "referenceCache.clear" or something? These are somewhat sketchy as they run a cleanup thread against their objects, so it's a bit dangerous to reassign.